tests/unit/src/test/scala/tests/FormattingLspSuite.scala

```diff
@@ -415,6 +415,50 @@ class FormattingLspSuite extends BaseLspSuite("formatting") {
     } yield ()
   }
 
+  test("rewrite-dialect-ignore-sbt") {
+    cleanWorkspace()
+    client.showMessageRequestHandler = { params =>
+      if (
+        params.getMessage().startsWith(Messages.UpdateScalafmtConf.beginning)
```

Comment:
Why do we need this if the `.scalafmt.conf` file doesn't change? Is it in order to make sure that the file will not change?

Could we instead throw an exception here if there are any params at all? No need for the new `beginning` value then.

Reply:
Changed it a bit to trigger rewrite. 
Now it checks that applied will be `scala212` instead of `scala212source3` that might come from sbt project.

Reply:
Could we not have just:
```
  params.getActions.asScala
          .find(_ == Messages.UpdateScalafmtConf.letUpdate)
```
without the `if` condition? We could get rid of `Messages.UpdateScalafmtConf.beginning`